tests/golos.publication_rewards_tests.cpp

```diff
@@ -880,4 +909,164 @@ BOOST_FIXTURE_TEST_CASE(golos_curation_test, reward_calcs_tester) try {
     check();
 } FC_LOG_AND_RETHROW()
 
+BOOST_FIXTURE_TEST_CASE(golos_delegators_test, reward_calcs_tester) try {
+    BOOST_TEST_MESSAGE("golos_delegators_test");
+    int64_t maxfp = std::numeric_limits<fixp_t>::max();
+    auto bignum = 500000000000;
+    init(bignum, 500000);
+    produce_blocks();
+    BOOST_TEST_MESSAGE("--- setrules");
+    using namespace golos_curation;
+    BOOST_CHECK_EQUAL(success(), setrules({"x", maxfp}, {golos_curation::func_str, maxfp}, {"1", bignum},
+        [](double x){ return x; }, golos_curation::func, [](double x){ return 1.0; }));
+    check();
+
+    BOOST_TEST_MESSAGE("--- add_funds_to_forum");
+    BOOST_CHECK_EQUAL(success(), add_funds_to_forum(50000));
+    check();
+
+    const uint8_t withdraw_intervals = 1;
+    const uint32_t withdraw_interval_seconds = 12;
+    const uint64_t vesting_min_amount = 0;
+    const uint64_t delegation_min_amount = 5e6;
+    const uint64_t delegation_min_remainder = 15e3;
+    const uint32_t delegation_min_time = 0;
+    const uint16_t delegation_max_interest = 10000;
+    const uint32_t delegation_return_time = 120;
+
+    auto vesting_withdraw = vest.withdraw_param(withdraw_intervals, withdraw_interval_seconds);
+    auto vesting_amount = vest.amount_param(vesting_min_amount);
+    auto delegation = vest.delegation_param(delegation_min_amount, delegation_min_remainder, delegation_min_time, delegation_max_interest, delegation_return_time);
+
+    auto params = "[" + vesting_withdraw + "," + vesting_amount + "," + delegation + "]";
+    BOOST_CHECK_EQUAL(success(), vest.set_params(_issuer, _token_symbol, params));
+
+    const int divider = vest.make_asset(1).get_amount();
+    const int min_remainder = delegation_min_remainder / divider;
+    const auto amount = vest.make_asset(min_remainder);
+    uint16_t interest_rate = 3000;
+    uint8_t interest_rate0 = 0;
```

Comment:
there is no need in this var, `0` can be used instead, it's value is quite straightforward and have no magic. (btw, it's better to use uint16_t for all percent variables)